pkg/operator/ceph/object/user.go

```diff
@@ -175,11 +175,18 @@ func UpdateUser(c *Context, user ObjectUser) (*ObjectUser, int, error) {
 
 // DeleteUser deletes the user with the given ID.
 func DeleteUser(c *Context, id string, opts ...string) (string, error) {
-	args := []string{"user", "rm", "--uid", id}
+	args := []string{"bucket", "list", "--uid", id}
```

Comment:
IMO it is better to do it other way around if rm fails then lists the bucket. Secondly the neat way to fix the issue which u are facing in https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/user/controller.go#L460 (parent function) than here either via listbucket(may need to add seperate fuction) or using s3 client(best way)

Reply:
Agreed, no need to list the buckets if the user deletion succeeds

Reply:
Made the changes in this file itself. After looking at the changes if you think that this still needs to be done in https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/user/controller.go#L460 Let me know I will do that.